prow/oss/terraform/modules/alerts/main.tf

```diff
@@ -12,6 +12,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+resource "google_monitoring_alert_policy" "pod-crashlooping" {
+  for_each     = var.prow_components
+  project      = var.project
+  display_name = "pod-crashlooping-${each.key}"
+  combiner     = "OR" # required
+
+  conditions {
+    display_name = "pod-crashlooping-${each.key}"
+
+    condition_monitoring_query_language {
+      # Alert if the service crashlooped, which results in more than 5 restarts in 5 minutes.
```

Comment:
These [k8s docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) indicate the crash loop backoff delay is used and increased after the first restart. The delay can reach 5 mins in which case we'd only get 1 restart per 5 mins.  So IIUC it would be impossible for this alert's condition to be met.

Reply:
I think you're right. I did see that doc but didn't compute it clearly.

event: restart, 10s ,restart,  20s ,restart,  40s ,restart,  80s ,restart,  160s ,restart,  300s ,restart

| time(s) | total restart on real crashlooping | restart in 10 minutes| total restart on master restart | restart in 10 minutes|
|-------|-----------|-------------|------------|--------------|
|0|1|1|1|1|
|10|2|2|2|2|
|30|3|3|3|3|
|70|4|4|4|4|
|150|5|5|5|5|
|310|6|5|6|5|
|330|6|4|6|4|
|370|6|3|6|3|
|450|6|2|6|2|
|610|7|2|6|1|
|630|7|1|6|0|
|670|7|1|6|0|
|750|7|1|6|0|
|910|7|1|6|0|

so in master restart case, there will be 1 restart in 10 minutes (0s to 610s), and 2 restarts in 440s(10s to 450s).
in real crashlooping case, there will be 2 restart in 10 minutes (0s to 610s), this might not be guaranteed depends on the resolution of the data and the cutoff of intervals.

I think it might be good to do either 1 restart in 12 minutes, or 2 restarts in 8 minutes. WDYT?

Reply:
We don't want to alert on a single restart. Maybe check for at least 1 restart per 6 minutes (checked every 6 minutes) for at least 12 minutes.  That would require at least 2 restarts in total so we'd avoid transient errors, but we shouldn't fire on any number of restarts within a 6 minute window as long as they are contained to that window.

Reply:
agreed, done